src/Ocelot/DependencyInjection/BuilderExtensions/IdentityServerExtensions.cs

```diff
@@ -0,0 +1,102 @@
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using IdentityServer4.AccessTokenValidation;
+using IdentityServer4.Models;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Ocelot.Configuration.Authentication;
+using Ocelot.Configuration.Creator;
+using Ocelot.Configuration.Provider;
+using Ocelot.Middleware;
+
+namespace Ocelot.DependencyInjection
+{
+    public static class OcelotBuilderExtensionsIdentityServer
+    {
+        public static void AddIdentityServer(this IOcelotBuilder builder)
```

Comment:
This should be private or internal because users should not be able to call it? Ocelot won't work without this being called?